php-checks/src/test/resources/checks/UnusedLocalVariableCheck.php

```diff
@@ -271,6 +271,11 @@ function compact_function() {
   return compact("a", "b");
 }
 
+function compact_function_with_backslash() {
+  $a = foo();
+  return \compact("a");
+}
+
```

Comment:
can we also detect the following? (it's still a FP - the call is recursive)
```
 $city  = "San Francisco"; // FP
 $location_vars = array("city");
 $result = compact($location_vars);
```

Reply:
[SONARPHP-809](https://jira.sonarsource.com/browse/SONARPHP-809) is about the backslash, and the [initially reported code](https://community.sonarsource.com/t/php-s1481-code-smell-unused-local-variables-should-be-removed-false-positive-with-compact/902) is covered by that fix. 

Your case seems to be less common, at least it was never reported by any user.  I believe that it should be handled as part of another ticket.

Reply:
Makes sense. I opened [SONARPHP-815](https://jira.sonarsource.com/browse/SONARPHP-815) the above mentioned FP